### PR TITLE
fix: demo extractor updated yara

### LINKED ISSUE #56: yara.Match unexpected incompatibility
Looks like 1.2.0 breaks compatibility for extractors that inspect yara rules.

Running maco with the demo extractors causes the following issue:

```
$ maco demo_extractors/ demo_extractors/
...
path: demo_extractors/complex/__pycache__/complex.cpython-312.pyc
run Complex extractor from rules ['Complex', 'ComplexAlt']
extractor error with demo_extractors/complex/__pycache__/complex.cpython-312.pyc ('StringMatch' object is not subscriptable)
Traceback (most recent call last):
  File ".../Maco/maco/cli.py", line 60, in process_file
    resp = collected.extract(stream, hits, extractor_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../Maco/maco/collector.py", line 118, in extract
    resp = extractor["module"]().run(stream, matches)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../Maco/demo_extractors/complex/complex.py", line 53, in run
  File "/home/seb/work/Maco/demo_extractors/complex/complex.py", line 53, in run
    strings = {y[2].decode("utf8") for x in matches for y in x.strings}
               ~^^^
TypeError: 'StringMatch' object is not subscriptable
...
```

### PR DESCRIPTION
**Pull request by seb-acsc**
This PR fixes the demo complex extractor for yara v4.3.0 compatibility and adds a test case so we can pick up the failure in the future.

Fixes #56

### CODE REVIEW

**Comment by seb-acsc** [comment]
unfortunately I can't see why the tests failed - is it related to the use of the `unittest` module?

**Comment by cccs-ml** [comment]
The tests are failing on Py3.8 and Py3.9.
Running locally the tests appear to give non-deterministic results on Py3.8 at least.

`-              'decoded_strings': ['Complex', 'Paradise'],
?                                  -----------

+              'decoded_strings': ['Paradise', 'Complex'],
?                                            +++++++++++`

Re-Running the tests results in some runs passing and others failing.

**Comment by cccs-rs** [comment]
We could convert those into sets in the test files, I believe we do that as a workaround in the AL pytests

File: demo_extractors/complex/complex.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def run(self, stream: BytesIO, matches: List[yara.Match]) -> Optional[model.Extr
         other = complex_utils.getdata()["result"]
         self.logger.debug("got data from lib")
         # example - accessing yara strings
-        strings = {y[2].decode("utf8") for x in matches for y in x.strings}
+        strings = {z.plaintext().decode("utf8") for x in matches for y in x.strings for z in y.instances}
```
Comment (cccs-ml) [inline demo_extractors/complex/complex.py:53]: I think if you change this to a `sorted([])` the test output would become deterministic and easier to test.

**Review by cccs-ml — CHANGES_REQUESTED** [review]

File: tests/test_demo_extractors.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                force=False,
+                include_base64=False,
+            )
+        self.assertEqual(
+            ret,
+            {
+                "Complex": {
+                    "family": "complex",
+                    "version": "5",
+                    "decoded_strings": ["Paradise", "Complex"],
```
Comment (cccs-ml) [inline tests/test_demo_extractors.py:39]: Change to: "decoded_strings": ["Complex", "Paradise"]
based on sorted output from previous comment.

File: demo_extractors/complex/complex.py (hunk 1 above)
Comment (cccs-ml) [inline demo_extractors/complex/complex.py:53]:
```suggestion
        strings = sorted({z.plaintext().decode("utf8") for x in matches for y in x.strings for z in y.instances})
```

File: tests/test_demo_extractors.py (hunk 2 above)
Comment (cccs-ml) [inline tests/test_demo_extractors.py:39]:
```suggestion
                    "decoded_strings": sorted(["Paradise", "Complex"]),
```

**Review by cccs-ml — CHANGES_REQUESTED** [review]
Forcing sorted output in the demo extractor would make testing easier.

**Review by cccs-rs — APPROVED** [review]

**Pull request merged**